sdk/cognitiveservices/azure-cognitiveservices-formrecognizer/DESIGN_DOC.md

```diff
@@ -0,0 +1,436 @@
+    List[List[TableCell]]
+    row_count: int
+    column_count: int
+
+class TableCell:
+    text: str
+    column_index: int
+    column_span: int
+    confidence: float
+    is_footer: bool
+    is_header: bool
+    row_index: int
+    row_span: int
+    bounding_box: List[float]
+    raw_field: List[ExtractedLine]
+
+class ExtractedLine:
+    text: str
+    bounding_box: List[float]
+    language: str
+    words: List[ExtractedWord]
+
+class ExtractedWord:
+    text: str
+    bounding_box: List[float]
+    confidence: float
+```
+
+### Layout Sample
+
+```python
+import pandas as pd
+from azure.ai.formrecognizer import FormRecognizerClient
+
+client = FormRecognizerClient(endpoint, credential)
+
+table_image = "https://i.stack.imgur.com/1FyIg.png"
+poller = client.begin_extract_layout(table_image)
+result = poller.result()
+
+dftable = pd.DataFrame(result[0].tables[0]) # page 1, table 1
+print(dftable)
+```
+
+## Custom
+
+The `CustomFormClient` provides all the operations necessary for training a custom model, analyzing a form with a 
+custom model, and managing a user's custom models on their account.
+
+The user can choose to train with or without labels using the methods `begin_labeled_training` or `begin_training`. 
+Both methods take as input a blob SAS uri or valid path to the documents to use for training. Each training method 
+will return a poller object which is used to get the training result.
+
+A custom model can be used to analyze forms using the `begin_extract_form` or `begin_extract_labeled_fields` methods.
+The `model_id` from the training result is passed into the methods, along with the input form to analyze (content-type
+is determined internally). Both methods return a poller object which is used to get the result object.
+
+In order for the user to manage their custom models, a few methods are available to list custom models, delete a model,
+and get a models summary for the account.
+
+### Custom Form Client
+```python
+from azure.ai.formrecognizer import CustomFormClient
+
+client = CustomFormClient(endpoint: str, credential: Union[CognitiveKeyCredential, TokenCredential])
+
+client.begin_labeled_training(
+    source: str, source_prefix_filter: str, include_sub_folders: bool=False
+) -> LROPoller -> LabeledCustomModel
+
+client.begin_training(
+    source: str, source_prefix_filter: str, include_sub_folders: bool=False
+) -> LROPoller -> CustomModel
+
+client.begin_extract_form(form: Any, model_id: str,) -> LROPoller -> List[ExtractedPage]
+
+client.begin_extract_labeled_fields(form: Any, model_id: str) -> LROPoller -> List[LabeledExtractedPage]
+
+client.list_custom_models() -> ItemPaged[ModelInfo]
+
+client.get_models_summary() -> ModelsSummary
+
+client.delete_custom_model(model_id: str) -> None
+```
+
+### Custom Models Unlabeled
+```python
+# Training ---------------------------------------------------
+class CustomModel:
+    model_id: str
+    status: str
+    created_date_time: ~datetime.datetime
```

Comment:
How are other SDKs naming their datetimes? Would it make sense to call this just `created` or `created_on`?

Reply:
Good point, a quick review of other SDKs shows that `created_on` would be more consistent naming.